docs/tsp/typeServerProtocol.ts

```diff
@@ -0,0 +1,383 @@
+        moduleName: string;
+    }
+
+    // Synthesized type information for a declaration that is not directly
+    // represented in the source code, but is derived from the declaration. Other languages
+    // may refer to this as an anonymous type.
+    export interface SynthesizedTypeInfo {
+        type: Type;
+
+        // An optional node that is not used by the type evaluator
+        // but can be used by language services to provide additional
+        // functionality (such as go-to-definition).
+        node?: Node;
+    }
+
+    // Declaration information for a symbol, which includes
+    // a list of declarations and potentially synthesized types for those declarations.
+    export interface SymbolDeclInfo {
+        decls: Declaration[];
+        synthesizedTypes: SynthesizedTypeInfo[];
+    }
+
+    // Describes an import
+    export interface ImportedModuleDescriptor {
+        leadingDots: number;
+        nameParts: string[];
+    }
+
+    // Options for resolving an alias declaration.
+    export interface ResolveAliasOptions {
+        resolveLocalNames?: boolean; // Whether to resolve local names in the alias declaration.
+        allowExternallyHiddenAccess?: boolean; // Whether to allow access to members that are hidden by external modules.
+        skipFileNeededCheck?: boolean; // Whether to skip checking if the file is needed for the alias resolution.
+    }
+
+    // Describes a file that's been opened in the language server which the type server needs to be aware of.
+    export interface TextDocumentOpenParams {
+        uri: string;
+        text: string;
+        // The version of the file, which is used to track changes to the file. It will be incremented each time the file is modified.
+        version: number;
+        // Optional: If the file is part of a chain of files, this can be used to indicate the URI of the
+        // file that is automatically 'imported' by this file. This is useful for cases like Jupyter notebooks
+        chainedFileUri?: string;
+    }
+
+    // Describes a file that's been closed in the language server which the type server needs to be aware of.
+    export interface TextDocumentCloseParams {
+        uri: string;
+    }
+
+    // Parameters for resolving an import
+    export interface ResolveImportParams {
+        // The URI of the source file where the import is referenced.
+        sourceUri: string;
+        // The descriptor of the imported module.
+        moduleDescriptor: ImportedModuleDescriptor;
+        snapshot: number;
+    }
+
+    // Represents settings that can be sent to the type server.
+
+    export type Settings = {
+        [key: string]: any;
+    };
+
+    // Requests and notifications for the type server protocol.
+    export enum Requests {
+        // First request to initialize the type server.
+        Initialize = 'typeServer/initialize',
+        // Request from client to shut down the type server.
+        Shutdown = 'typeServer/shutdown',
```

Comment:
Nit: "Shutdown" is an adjective, whereas "shut down" is a verb. It's analogous to `logout` and `log out`. I think you mean the latter, so it should be `ShutDown`.